vowpalwabbit/cb_adf.cc

```diff
@@ -567,7 +567,7 @@ base_learner* cb_adf_setup(options_i& options, vw& all)
 
   auto base = as_multiline(setup_base(options, all));
   all.p->lp = CB::cb_label;
-  all.label_type = label_type_t::cb;
+  all.gs.label_type = label_type_t::cb;
```

Comment:
@jackgerrits should this be encapsulated on it's own function? seems like in a lot of places we do the same operation of p->lp and label_type. Could even be like a state machine in case there's more to do when changing this.

Reply:
see this commit for all cases https://github.com/VowpalWabbit/vowpal_wabbit/pull/2483/commits/529752434cb171175caad38b765a52e9f21ecaac

Reply:
Yeah encapsulating this could be good! I agree